contracts/modules/capital/Ramm.sol

```diff
@@ -441,21 +454,54 @@ contract Ramm is IRamm, MasterAwareV2, ReentrancyGuard {
     return timestamp.divCeil(PERIOD_SIZE) % GRANULARITY;
   }
 
-  function getObservation(
+  function calculateTimeOnRatchetAndBV(
+    State memory previousState,
+    uint timeElapsed,
+    uint stateRatchetSpeed,
+    uint supply,
+    uint capital,
+    bool isAbove
+  ) internal pure returns (uint timeOnRatchet, uint timeOnBV) {
+
+    // innerAbove = (eth * supply) - (buffer * capital * nxm)
+    // innerBelow = (buffer * capital * nxm) - (eth * supply)
+
+    uint prevNxm = isAbove ? previousState.nxmA : previousState.nxmB;
+    uint bufferMultiplier = isAbove ? (PRICE_BUFFER_DENOMINATOR + PRICE_BUFFER) : (PRICE_BUFFER_DENOMINATOR - PRICE_BUFFER);
+
+    uint ethLiquidity = previousState.eth * supply;
+    uint nxmLiquidity = (bufferMultiplier * capital * prevNxm) / PRICE_BUFFER_DENOMINATOR;
```

Comment:
are `ethLiquidity` and `nxmLiquidity` the correct variable names for these values?